classes/controller/AdminController.php

```diff
@@ -2733,12 +2733,16 @@ protected function l($string, $class = null, $addslashes = false, $htmlentities
     {
         @trigger_error(__FUNCTION__ . 'is deprecated. Use AdminController::trans instead.', E_USER_DEPRECATED);
 
-        $parameters = [];
-        if ($htmlentities) {
-            $parameters['legacy'] = 'htmlspecialchars';
+        if ($htmlentities === true) {
+            @trigger_error(
+                'Use htmlentities parameter will be removed in version 9.0, escaping translation is not the translator responsability.',
+                E_USER_DEPRECATED
+            );
+
+            return htmlspecialchars($this->translator->trans($string, []), ENT_NOQUOTES);
         }
 
-        return $this->translator->trans($string, $parameters);
+        return $this->translator->trans($string, []);
```

Comment:
You're removing the replacement tokens here, you need to keep them.
The problem can be solved more easily in this function because it already had a dedicated $htmlentities parameter

```php
if ($htmlentities === true) {
    return htmlspecialchars($this->translator->trans($string, $parameters), ENT_NOQUOTES);
}

return $this->translator->trans($string, $parameters);
```